src/github/github.controller.ts

```diff
@@ -0,0 +1,225 @@
+    @Query('page') page: number = 1,
+    @Query('limit') limit: number = 20,
+  ) {
+    try {
+      const repos = await this.githubService.listRepos(
+        req,
+        organizationName,
+        page,
+        limit,
+      );
+      return repos;
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get('oauth/redirect')
+  githubAuthRedirect() {
+    const clientId = Env.GITHUB.CLIENT_ID;
+    const githubAuthUrl = `https://github.com/login/oauth/authorize?client_id=${clientId}&scope=repo,read:user,read:org`;
+    return githubAuthUrl;
+  }
+
+  @Get('app/redirect')
+  async redirectToGitHubAppInstallation() {
+    const githubAppUrl = `https://github.com/apps/${Env.GITHUB.APP_NAME}/installations/select_target`;
+    return githubAppUrl;
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Post('app/install')
+  async handleInstallation(
+    @Req() req: TRequest,
+    @Query('installationId') installationId: string,
+  ) {
+    try {
+      const result = await this.githubService.handleInstallation(
+        req,
+        installationId,
+      );
+      return result;
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Get('user')
+  async getUser(@Req() req: TRequest) {
+    try {
+      const result = await this.githubService.getUser(req);
+      return result;
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Get(':repoName/practices')
+  async getPractices(
+    @Req() req: TRequest,
+    @Param('repoName') repoName: string,
+  ) {
+    try {
+      return await this.githubService.getPractices(req, repoName);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Put(':repoName/practices')
+  async updatePractices(
+    @Req() req: TRequest,
+    @Param('repoName') repoName: string,
+    @Body() updatePracticesDto: UpdatePracticesDto,
+  ) {
+    try {
+      return await this.githubService.updatePractices(
+        req,
+        repoName,
+        updatePracticesDto,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
```

Comment:
PullProofAI

Similar to other methods, the `updatePractices` method would benefit from more detailed error handling to handle different failure scenarios effectively.

```diff
+       // Add specific error handling logic for various failure cases
```


<!-- This is an auto-generated comment by PullProofAI -->